controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -288,6 +293,74 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}, nil
 }
 
+func (r *HCPVaultSecretsAppReconciler) startOrphanedShadowSecretCleanup(ctx context.Context) error {
+	var err error
+
+	r.once.Do(func() {
+		for {
+			select {
+			case <-ctx.Done():
+				if ctx.Err() != nil {
+					err = ctx.Err()
+				}
```

Comment:
It doesn't look like we are breaking/returning here? I think this might lead to a Go routine being leaked.